docs/src/plotting_functions/violin.md

```diff
@@ -17,22 +17,50 @@ ys = randn(1000)
 violin(xs, ys)
 ```
 
+
+```@example
+using CairoMakie
+CairoMakie.activate!() # hide
+AbstractPlotting.inline!(true) # hide
+
+xs = rand(1:3, 1000)
+ys = map(xs) do x
+    return x == 1 ? randn() : x == 2 ? 0.5 * randn() : 5 * rand()
+end
+
+violin(xs, ys, datalimits = extrema)
+```
+
+
 ```@example
 using CairoMakie
 CairoMakie.activate!() # hide
 AbstractPlotting.inline!(true) # hide
 
-xs1 = rand(1:3, 1000)
-ys1 = randn(1000)
-dodge1 = rand(1:2, 1000)
+N = 1000
+xs = rand(1:3, N)
+dodge = rand(1:2, N)
+side = rand([:left, :right], N)
+color = Observable((left = :orange, right = :teal))
```

Comment:
Passing directly the named tuple, ie `color = (left = :orange, right = :teal)` causes a bug with the theme merging mechanism, that is:

```julia
julia> violin(xs, ys, side = side, color = (left=:orange, right=:teal))
ERROR: Type missmatch while merging plot attributes with theme for key: color.
Found Observable{Any} with 0 listeners. Value:
:black in theme, while attributes contains: Attributes with 2 entries:
  left => orange
  right => teal
```

I imagine that should be fixed independently in AbstractPlotting?